Lib/functools.py

```diff
@@ -193,6 +193,65 @@ def total_ordering(cls):
     return cls
 
 
+################################################################################
+### topological sort
+################################################################################
+
+class CycleError(Exception):
```

Comment:
This ought to inherit from `ValueError`, since it indicates an inappropriate argument value (of the correct type).